packages/seacas/libraries/ioss/src/Ioss_Beam2.C

```diff
@@ -79,13 +79,15 @@ int Ioss::Beam2::number_nodes_face(int face) const
 {
   // face is 1-based.  0 passed in for all faces.
   assert(face >= 0 && face <= number_faces());
+  (void)face;
```

Comment:
Can we use a macro or something here to make it more visible what is happening -- UNUSED(face) or something like that.  Then can tailor for different compilers as I think there ar some that still warn even with `(void)face;`.  Can use C++ attributes `[[maybe_unused]]` (?) in the macro...  

I know macros are frowned upon, but makes it easier to tailor for multiple compilers...

Reply:
That's a C++17 attribute (and belongs on the parameter itself IIRC). Where should this macro live?

Reply:
That works.  Or `Ioss_CodeTypes.h`

Reply:
Is Kitware using C++17 yet?  Sierra has moved to C++17 and Trilinos will soon, so C++17 will soon start appearing in the source code.  I can delay if that will adversely affect your builds though.

Reply:
VTK is still firmly in C++11. I think VTK-m will push to C++14 in the short term. Our vendoring has a commit that makes things C++11-compatible though; it's not too much  a deal as long as that match doesn't get too crazy. https://gitlab.kitware.com/third-party/seacas/-/commit/42fa9767d657ff14ca2b9134cfa3d335aea5323f